### PR TITLE
Promotional Flair for external Billboard

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Adding a promoted flair when the billboard is an "external" billboard.

## Related Tickets & Documents

- Related Issue #https://github.com/forem/forem-internal-eng/issues/558
- Closes #https://github.com/forem/forem-internal-eng/issues/558

## QA Instructions, Screenshots, Recordings

Screenshot 2023-03-10 at 13 39 57

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: These are ui changes and display ads do not currently have tests in cypress, however I think we should be testing the UI interactions before we get too far a long and so I'm planning on spending some time thinking about it. The randomness of Display Ads on the UI make it difficult to test.  
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by Ridhwana** [comment]
Waiting for https://github.com/forem/forem/pull/19208 to be merged before reviewing and approving this PR.

File: app/views/shared/_display_ad.html.erb [hunk 1]
```diff
@@ -5,17 +5,22 @@
   data-context-type="<%= data_context_type %>"
   data-type-of="<%= display_ad.type_of %>">
   <div class="crayons-sponsorship__header relative">
-    <% if Page.exists?(slug: "billboards") && display_ad.organization_id? && display_ad.type_of == "community" %>
+    <% if display_ad.organization_id? && (display_ad.type_of == "community" || display_ad.type_of == "external") %>
       <img width="24" height="24" class="radius-default crayons-sponsorship__image" src="<%= display_ad.organization.profile_image_url_for(length: 64) %>" alt="profile" loading="lazy" />
-      <div class="crayons-sponsorship__title pl-1 fs-s"><%= display_ad.organization.name %></div>
+      <div class="crayons-sponsorship__title pl-1 fs-s fw-medium"><%= display_ad.organization.name %></div>
+      <% if display_ad.type_of == "external" %>
+        <span class="crayons-sponsorship__indicator c-indicator c-indicator--subtle c-indicator--round fs-xs fw-medium ml-1 py-1 px-2">PROMOTED</span>
```
Comment (rt4914) [inline app/views/shared/_display_ad.html.erb:12]: Optional suggestion: Maybe we can define this `PROMOTED` in en.yml and fr.yml files.

**Review by rt4914 — APPROVED** [review]

**Review by jaw6 — APPROVED** [review]

**Comment by Ridhwana** [comment]
Just a note that the UI is a bit broken at the moment as I merged main in, I think https://github.com/forem/forem/pull/19206 may have changed some padding that affects the way Promoted is displayed.

**Comment by Ridhwana** [comment]
Ok the latest commits have now fixed the style issues: 

Screenshot 2023-03-22 at 12 55 20

Screenshot 2023-03-22 at 12 55 40

@rt4914 and @jaw6 not sure if you want to look at the changes again and give a thumbs up?

File: app/views/shared/_display_ad.html.erb (hunk 1 above)
Comment (Ridhwana) [reply app/views/shared/_display_ad.html.erb]: Done ✅ thanks for the suggestion!

File: app/assets/stylesheets/widgets.scss [hunk 2]
```diff
@@ -348,8 +353,8 @@
 
 .crayons-layout__comments-display-ad {
   .crayons-sponsorship {
-    .crayons-sponsorship__title {
-      padding-top: var(--su-5);
```
Comment (Ridhwana) [inline app/assets/stylesheets/widgets.scss:352]: I moved this a utility class in the html

File: app/assets/stylesheets/widgets.scss [hunk 3]
```diff
@@ -348,8 +353,8 @@
...
+    .c-indicator.crayons-sponsorship__indicator {
+      line-height: 150%;
```
Comment (Ridhwana) [inline app/assets/stylesheets/widgets.scss:357]: I needed to add some specificity here so that the line-height specified in the Figma would pull through.

**Review by jaw6 — APPROVED** [review]

**Review by rt4914 — APPROVED** [review]
Looks good. Thanks @Ridhwana

**Comment by benhalpern** [comment]
Looking at this in prod, padding above the org name looks off now. I think that `pt-5` should just be `pt-1` and it would work.

Screen Shot 2023-03-23 at 5 55 16 PM
Screen Shot 2023-03-23 at 5 55 08 PM

The appearance of padding within the flair element also seems off. There might be a few levers to pull here, but adjusting the line height to 15px appears to fix the issue.

There may be additional wac-a-mole to play to get the styling pixel perfect, but I think this deserves whatever quick fix is possible to get this looking good in prod for the moment.

**Pull request merged**